common/lib/client/rest.ts

```diff
@@ -0,0 +1,225 @@
+import Platform from 'platform';
+import * as Utils from '../util/utils';
+import Logger, { LoggerOptions } from '../util/logger';
+import Defaults from '../util/defaults';
+import Auth from './auth';
+import Push from './push';
+import Http from 'platform-http';
+import PaginatedResource, { HttpPaginatedResponse, PaginatedResult } from './paginatedresource';
+import Channel from './channel';
+import ErrorInfo from '../types/errorinfo';
+import Stats from '../types/stats';
+import HttpMethods from '../../constants/HttpMethods';
+import { ChannelOptions } from '../../types/channel';
+import { PaginatedResultCallback, StandardCallback } from '../../types/utils';
+import { ErrnoException, RequestParams } from '../../types/http';
+import ClientOptions, { DeprecatedClientOptions, NormalisedClientOptions } from '../../types/ClientOptions';
+
+const noop = function() {};
+const msgpack = Platform.msgpack;
+
+class Rest {
+	options: NormalisedClientOptions;
+	baseUri: (host: string) => string;
+	authority: (host: string) => string;
+	_currentFallback: null | {
+		host: string,
+		validUntil: number,
+	};
+	serverTimeOffset: number | null;
+	auth: Auth;
+	channels: Channels;
+	push: Push;
+
+	constructor(options: ClientOptions | string) {
+		/* normalise options */
```

Comment:
I think this comment is not linked with code provided in the next line

Reply:
Yeah it's definitely not helpful, I've removed it in 73294d4